map/src/main/kotlin/edu/artic/map/MapFragment.kt

```diff
@@ -218,6 +231,23 @@ class MapFragment : BaseViewModelFragment<MapViewModel>() {
                             }
                         }
                     }
+                    is ArticObject -> {
+                        val mapObject = marker.tag as ArticObject
```

Comment:
all this business logic should be in the ViewModel

Reply:
Updated.